e2e/cypress/support/hooks.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// Updates the config with the provided values before the test suite and then resets the config to the previous value
+// after the test suite has finished.
+export function testWithConfig(config) {
+    before(() => {
+        let originalConfig;
+        cy.apiGetConfig().then((resp) => {
+            originalConfig = resp.body;
+        });
+
+        cy.apiUpdateConfig(config);
+
+        after(() => {
+            cy.apiUpdateConfig(originalConfig);
+        });
+    });
+}
```

Comment:
We usually avoid cleaning up state after tests, (reference: https://docs.cypress.io/guides/references/best-practices.html#Using-after-or-afterEach-hooks) but for system config, we mostly encountered flaky test due to unexpected setting.  With that, I'm all in with this approach and will update other parts to have this implemented once merged on master. :100: